### PR TITLE
[WIP] Use optional config groups in Hibernate ORM configuration

### LINKED ISSUE #5608: Hibernate ORM should use new optional config groups
**Description**
After #5387 is merged, we gain support for optional config groups.  Optional config groups, as explained in the docs, allow a config group _as a whole_ to be considered optional, which also implies that the required properties in the group are _only_ required when _any other_ property within the group or its nested children are specified.

This should fit the Hibernate ORM configuration well.  Presently the Hibernate ORM configuration uses many individually-Optional properties with a query method typically named `isAnyPropertySet`, coupled with manual validation of combinations of properties.  This extra validation step is ripe to be omitted, with a resultant savings in code complexity.  Instead, it can simply query for the presence of the various configuration groups to determine if any of the corresponding properties have been set.

/cc @Sanne

### PR DESCRIPTION
**Pull request by gwenneg**
Fixes #5608

### CODE REVIEW

**Comment by gwenneg** [comment]
This was on a branch for a while, I'll finish it during the week-end.

I tried to get rid of `HibernateOrmConfig.isAnyPropertySet()` a few weeks ago but it didn't end well. I'll try again and then promote the PR for a review.

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfig.java [hunk 1]
```diff
@@ -78,40 +78,40 @@
     /**
      * The size of the batches used when loading entities and collections.
      *
-     * `-1` means batch loading is disabled. This is the default.
+     * `0` means batch loading is disabled. This is the default.
      *
      * @asciidoclet
      */
-    @ConfigItem(defaultValue = "-1")
```
Comment (gwenneg) [inline extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfig.java]: This needs extra attention. I don't see any reason to use `-1` here but maybe I missed something.

**Comment by gwenneg** [comment]
I'll fix the build failure (which is certainly related) tonight.

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfig.java (hunk 1 above)
Comment (dmlloyd) [reply extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfig.java]: Maybe it should be an `OptionalInt`?

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfig.java [hunk 2]
```diff
@@ -131,10 +131,10 @@ public boolean isAnyPropertySet() {
...
                 batchFetchSize > 0 ||
                 statistics ||
-                query.isAnyPropertySet() ||
-                database.isAnyPropertySet() ||
-                jdbc.isAnyPropertySet() ||
-                log.isAnyPropertySet() ||
+                query.isPresent() ||
+                database.isPresent() ||
+                jdbc.isPresent() ||
+                log.isPresent() ||
```
Comment (dmlloyd) [inline extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfig.java]: I suppose the next step would be to see if we can make some of the properties in the optional groups be individually non-optional.

**Comment by gwenneg** [comment]
The failures are caused by the optional config groups not being populated while it seems they should be. I'll dig this further tomorrow.

**Comment by gsmet** [comment]
This is out of date. Please open a new PR if you want to pursue this.

**Pull request closed without merging**